metals/src/main/scala/scala/meta/internal/metals/debug/DebugProvider.scala

```diff
@@ -256,12 +258,8 @@ class DebugProvider(
             socket
           }
 
-        val connWithTimeout =
-          // if slow task is supported users can stop it themselves
-          if (clientConfig.slowTaskIsOn()) conn
-          else conn.withTimeout(60, TimeUnit.SECONDS)
-
-        connWithTimeout
+        conn
+          .withTimeout(60, TimeUnit.SECONDS)
```

Comment:
The logic in here was: if client supports slowTask (thus cancel) do not add a timeout to creating connection (user can cancel it), otherwise add timeout. Now we switch to  lsp `workDoneProgress`. Problem is that even if client supports LSP `workDoneProgress` it doesn't mean they support cancel for `workDoneProgress`, LSP specification basically says that client can ignore the cancellation option in `workDoneProgress` if it's not supported. So I decided that the safest option is to always add the timeout.

Reply:
ahh ok that makes sense. 👍🏼